web/src/components/profile/ProfileNewJerseyAddress.tsx

```diff
@@ -52,87 +56,97 @@ export const ProfileNewJerseyAddress = (): ReactElement => {
+    return business && isStartingBusiness(business) && business.formationData.completedFilingPayment
+      ? true
+      : false;
+  };
+
   return (
     <>
-      <div className="margin-y-4">
-        <div id={`question-addressLine1`} className="text-field-width-default add-spacing-on-ele-scroll">
-          <AddressTextField
-            label={Config.formation.fields.addressLine1.label}
-            fieldName="addressLine1"
-            validationText={getFieldErrorLabel("addressLine1")}
-            className={"margin-bottom-2"}
-            errorBarType="ALWAYS"
-            onValidation={onValidation}
-          />
-        </div>
-        <div id={`question-addressLine2`} className="text-field-width-default add-spacing-on-ele-scroll">
-          <AddressTextField
-            label={Config.formation.fields.addressLine2.label}
-            secondaryLabel={Config.formation.general.optionalLabel}
-            errorBarType="ALWAYS"
-            fieldName="addressLine2"
-            validationText={getFieldErrorLabel("addressLine2")}
-            className="margin-bottom-2"
-            onValidation={onValidation}
-          />
-        </div>
-        <div className="text-field-width-default">
-          <WithErrorBar
-            hasError={doSomeFieldsHaveError(["addressState", "addressZipCode", "addressMunicipality"])}
-            type="DESKTOP-ONLY"
-          >
-            <div className="grid-row tablet:grid-gap-2">
-              <div className="grid-col-12 tablet:grid-col-6">
-                <WithErrorBar hasError={doesFieldHaveError("addressMunicipality")} type="MOBILE-ONLY">
-                  <span className="text-bold">{Config.formation.fields.addressCity.label}</span>
-                  <AddressMunicipalityDropdown onValidation={onValidation} />
-                </WithErrorBar>
-              </div>
-              <div className="grid-col-12 tablet:grid-col-6">
-                <WithErrorBar
-                  hasError={doSomeFieldsHaveError(["addressState", "addressZipCode"])}
-                  type="MOBILE-ONLY"
-                >
-                  <div className="grid-row grid-gap tablet:grid-gap-2">
-                    <div className="grid-col-6">
-                      <strong>
-                        <ModifiedContent>{Config.formation.fields.addressState.label}</ModifiedContent>
-                      </strong>
-                      <div
-                        id={`question-addressState`}
-                        className="text-field-width-default add-spacing-on-ele-scroll"
-                      >
-                        <StateDropdown
-                          fieldName="addressState"
-                          value={"New Jersey"}
-                          validationText={Config.formation.fields.addressState.error}
-                          disabled={true}
-                          onSelect={(): void => {}}
-                        />
+      {areAddressFieldsDisabled() ? (
```

Comment:
[sand] This function name is not consistent with the way we name checks like this. Because of the mental tax of evaluating this function name, we generally make these "positive".

Personal preference, while it's not grammatically correct I'd say "is" over "are". There are multiple fields, but `isCondition` is a better visual indicator of the function's purpose. I'd also argue that we're talking about a condition that affects a single component making a singular check more appropriate.

`isAddressFieldsEnabled`

Reply:
Updated